### PR TITLE
Add recipe for valet

### PR DESCRIPTION
**Pull request by bebatut**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by bebatut** [comment]
I found an issue when testing it locally. I am fixing it

File: recipes/valet/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+mkdir -p $PREFIX/bin
+
+cp src/py/* $PREFIX/bin
+export VALET=$PREFIX/bin
```
Comment (bgruening) [inline recipes/valet/build.sh]: Is this export needed at runtime?

Comment (bebatut) [reply recipes/valet/build.sh]: yes. Is there a different way to do that?

Comment (bgruening) [reply recipes/valet/build.sh]: This is currently only valid in the build environment.
If you would have a single binary, I would recommend a wrapper. E.g. a file called `_binary` that is a shell script that `export VLATE=`which binary` && binary`.
In your case it is probably better to patch the python scripts to automatically set VALET to `os.path.realpath(__file__)`

Comment (bebatut) [reply recipes/valet/build.sh]: :) 
I was testing the second option

**Comment by bebatut** [comment]
I fixed most of the issues locally. But I still have 
```
dyld: Library not loaded: @rpath/libintl.9.dylib
  Referenced from: /Users/bebatut/miniconda3/envs/valet/lib/R/lib/libR.dylib
  Reason: image not found
```

**Comment by bgruening** [comment]
Fixed and green! Thanks!

**Pull request merged**